react/features/desktop-picker/functions.ts

```diff
@@ -1,70 +1,49 @@
-import { IReduxState } from '../app/types';
-
 import logger from './logger';
 import { ElectronWindowType } from './types';
 
-/**
- * Returns root conference state.
- *
- * @param {IReduxState} state - Global state.
- * @returns {Object} Conference state.
- */
-export const getDesktopPicker = (state: IReduxState) => state['features/desktop-picker'];
-
-/**
-* Selector to return a list of knocking participants.
-*
-* @param {IReduxState} state - State object.
-* @returns {IDesktopSources}
-*/
-export function getDesktopPickerSources(state: IReduxState) {
-    const root = getDesktopPicker(state);
-
-    return root.sources;
-}
-
 
 /**
  * Begins a request to get available DesktopCapturerSources.
  *
- * @param {Array} types - An array with DesktopCapturerSource type strings.
  * @param {Object} options - Additional configuration for getting a list of
  * sources.
+ * @param {Array} options.types - An array with DesktopCapturerSource type strings.
  * @param {Object} options.thumbnailSize - The desired height and width of the
  * return native image object used for the preview image of the source.
  * @returns {Function}
  */
-export function obtainDesktopSources(types: string[], options: { thumbnailSize?: Object; } = {}) {
-    const capturerOptions: any = {
-        types
-    };
-
-    if (options.thumbnailSize) {
-        capturerOptions.thumbnailSize = options.thumbnailSize;
-    }
-
-    return new Promise((resolve, reject) => {
-        const { JitsiMeetElectron } = window as ElectronWindowType;
+export function obtainDesktopSources(options: { thumbnailSize?: Object; types: string[]; }) {
+    const { JitsiMeetElectron } = window as ElectronWindowType;
 
-        if (JitsiMeetElectron?.obtainDesktopStreams) {
+    // TODO: delete this after 2 releases
+    if (JitsiMeetElectron?.obtainDesktopStreams) {
+        return new Promise((resolve, reject) => {
             JitsiMeetElectron.obtainDesktopStreams(
                 (sources: Array<{ id: string; }>) => resolve(_separateSourcesByType(sources)),
                 (error: Error) => {
                     logger.error(
                         `Error while obtaining desktop sources: ${error}`);
                     reject(error);
                 },
-                capturerOptions
+                options
             );
-        } else {
-            const reason = 'Called JitsiMeetElectron.obtainDesktopStreams'
-                + ' but it is not defined';
+        });
+    }
 
-            logger.error(reason);
+    if (APP.API.requestDesktopSources) {
```

Comment:
So it is not clear for me how this would be missing? I mean the APP.API.requestDestopSources is part of app.bundle.js and this code as well therefore you either have both of them or you don't have any of them. Am I missing something?

Reply:
Yes, remember the case of Jitsi Meet electron with a bundled External API.

The good news is after the change to reject unknown requests that will no longer be a problem so we'll wait a couple of releases and remove the workarounds.

Reply:
I think @hristoterezov is right. requestDesktopSources is on the API.js side and we can be sure it is included in bundle

Reply:
Ah yes yes, correct.